tests/unit/libc_wrapper_test.cc

```diff
@@ -0,0 +1,55 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ * Copyright (C) 2024 ScyllaDB.
+ */
+#include <coroutine>
+#include <iostream>
+#include <grp.h>
+
+#include <seastar/core/coroutine.hh>
+#include <seastar/core/reactor.hh>
+#include <seastar/core/seastar.hh>
+
+#include <seastar/testing/test_case.hh>
+
+using namespace seastar;
+
+SEASTAR_TEST_CASE(getgrnam_error_test) {
+    size_t buflen = 1;
+    char buf[1];
+    memset(&buf, 0, buflen);
+    BOOST_REQUIRE_THROW(co_await getgrnam("root", buf, buflen), std::system_error);
+}
+
+SEASTAR_TEST_CASE(getgrnam_group_name_does_not_exist_test) {
+    size_t buflen = 1024;
+    char buf[1024];
+    memset(&buf, 0, buflen);
```

Comment:
Again, I think all the memset() calls in this file are unnecessary, but please correct me if I'm wrong.